src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/ClusterAwareConnectionProxy.java

```diff
@@ -905,9 +913,10 @@ protected void failoverReader(int failedHostIdx) throws SQLException {
       this.failoverStartTimeMs = 0;
     }
 
-    if (result == null || !result.isConnected()) {
+    if (!result.isConnected()) {
```

Comment:
I'd keep the previous code. It's safer.

Reply:
both the failover procedures don't return null anymore; they always return a failed or successful result. But if you want me to add the check back in case we change the implementation in the future then I can add it back

Reply:
I'd prefer it back.